drivers/sensor/icm42605/icm42605.c

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright (c) 2020 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT invensense_icm42605
+
+#include <drivers/spi.h>
+#include <init.h>
+#include <sys/byteorder.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+
+#include "icm42605.h"
+#include "icm42605_reg.h"
+#include "icm42605_setup.h"
+#include "icm42605_spi.h"
+
+LOG_MODULE_REGISTER(ICM42605, CONFIG_SENSOR_LOG_LEVEL);
+
+/* see "Accelerometer Measurements" section from register map description */
+static void icm42605_convert_accel(struct sensor_value *val, int16_t raw_val,
+				   uint16_t sensitivity_shift)
+{
+	int64_t conv_val;
+
+	conv_val = ((int64_t)raw_val * SENSOR_G) >> sensitivity_shift;
+	val->val1 = conv_val / 1000000;
+	val->val2 = conv_val % 1000000;
+}
+
+/* see "Gyroscope Measurements" section from register map description */
+static void icm42605_convert_gyro(struct sensor_value *val, int16_t raw_val,
+				  uint16_t sensitivity_x10)
+{
+	int64_t conv_val;
+
+	conv_val = ((int64_t)raw_val * SENSOR_PI * 10) /
+		   (sensitivity_x10 * 180U);
+	val->val1 = conv_val / 1000000;
+	val->val2 = conv_val % 1000000;
+}
+
+/* see "Temperature Measurement" section from register map description */
+static inline void icm42605_convert_temp(struct sensor_value *val,
+					 int16_t raw_val)
+{
+	val->val1 = (((int64_t)raw_val * 100) / 207) + 25;
+	val->val2 = ((((int64_t)raw_val * 100) % 207) * 1000000) / 207;
+
+	if (val->val2 < 0) {
+		val->val1--;
+		val->val2 += 1000000;
+	} else if (val->val2 >= 1000000) {
+		val->val1++;
+		val->val2 -= 1000000;
+	}
+}
+
+static int icm42605_channel_get(struct device *dev,
```

Comment:
The device struct should be const here and throughout the driver

Reply:
Fixed.

Reply:
You fixed it here, but there are other instances in the driver that still need to be fixed

Reply:
OK, I will check all functions.

Reply:
@MaureenHelm Fixed other instances also.

Reply:
Many are fixed, thank you. But you need to move these fixes to the commit `drivers: sensor: Initial driver of Invensense ICM42605`.

There are also a few remaining that still need to be fixed.

Reply:
@MeganHansen Look like commit issue of my side.
Will push correct fix and will double check

Reply:
@MaureenHelm Fixed.